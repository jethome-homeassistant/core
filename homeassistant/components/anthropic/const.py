"""Constants for the Anthropic integration."""

import logging

DOMAIN = "anthropic"
LOGGER = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "Claude conversation"

CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "claude-3-5-haiku-latest"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 3000
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
CONF_THINKING_BUDGET = "thinking_budget"
RECOMMENDED_THINKING_BUDGET = 0
MIN_THINKING_BUDGET = 1024
CONF_WEB_SEARCH = "web_search"
RECOMMENDED_WEB_SEARCH = False
CONF_WEB_SEARCH_USER_LOCATION = "user_location"
RECOMMENDED_WEB_SEARCH_USER_LOCATION = False
CONF_WEB_SEARCH_MAX_USES = "web_search_max_uses"
RECOMMENDED_WEB_SEARCH_MAX_USES = 5
CONF_WEB_SEARCH_CITY = "city"
CONF_WEB_SEARCH_REGION = "region"
CONF_WEB_SEARCH_COUNTRY = "country"
CONF_WEB_SEARCH_TIMEZONE = "timezone"

NON_THINKING_MODELS = [
    "claude-3-5",  # Both sonnet and haiku
    "claude-3-opus",
    "claude-3-haiku",
]

WEB_SEARCH_UNSUPPORTED_MODELS = [
    "claude-3-haiku",
    "claude-3-opus",
    "claude-3-5-sonnet-20240620",
    "claude-3-5-sonnet-20241022",
]
