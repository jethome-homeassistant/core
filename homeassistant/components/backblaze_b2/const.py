"""Constants for the Backblaze B2 integration."""

from collections.abc import Callable
from typing import Final

from homeassistant.util.hass_dict import HassKey

DOMAIN: Final = "backblaze_b2"

CONF_KEY_ID = "key_id"
CONF_APPLICATION_KEY = "application_key"
CONF_BUCKET = "bucket"
CONF_PREFIX = "prefix"

DATA_BACKUP_AGENT_LISTENERS: HassKey[list[Callable[[], None]]] = HassKey(
    f"{DOMAIN}.backup_agent_listeners"
)

METADATA_FILE_SUFFIX = ".metadata.json"
METADATA_VERSION = "1"

BACKBLAZE_REALM = "production"
