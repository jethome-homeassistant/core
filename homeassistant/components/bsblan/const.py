"""Constants for the BSB-Lan integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

# Integration domain
DOMAIN: Final = "bsblan"

LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(seconds=12)  # Legacy interval, kept for compatibility
SCAN_INTERVAL_FAST = timedelta(seconds=12)  # For state/sensor data
SCAN_INTERVAL_SLOW = timedelta(minutes=5)  # For config data

# Services
DATA_BSBLAN_CLIENT: Final = "bsblan_client"

ATTR_TARGET_TEMPERATURE: Final = "target_temperature"
ATTR_INSIDE_TEMPERATURE: Final = "inside_temperature"
ATTR_OUTSIDE_TEMPERATURE: Final = "outside_temperature"

CONF_PASSKEY: Final = "passkey"

DEFAULT_PORT: Final = 80
