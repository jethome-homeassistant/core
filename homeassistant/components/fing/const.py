"""Const for the Fing integration."""

DOMAIN = "fing"
UPNP_AVAILABLE = "upnp_available"
