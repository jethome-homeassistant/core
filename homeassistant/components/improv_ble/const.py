"""Constants for the Improv BLE integration."""

from __future__ import annotations

import asyncio

from homeassistant.util.hass_dict import HassKey

DOMAIN = "improv_ble"

PROVISIONING_FUTURES: HassKey[dict[str, asyncio.Future[str]]] = HassKey(DOMAIN)

# Timeout in seconds to wait for another integration to register a next flow
# after successful provisioning (e.g., ESPHome discovering the device)
PROVISIONING_TIMEOUT = 10.0
