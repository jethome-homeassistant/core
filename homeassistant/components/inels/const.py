"""Constants for the iNELS integration."""

import logging

from homeassistant.const import Platform

DOMAIN = "inels"
TITLE = "iNELS"

PLATFORMS: list[Platform] = [
    Platform.SWITCH,
]

LOGGER = logging.getLogger(__package__)
