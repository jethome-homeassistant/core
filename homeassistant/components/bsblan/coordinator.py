"""DataUpdateCoordinator for the BSB-Lan integration."""

from dataclasses import dataclass
from datetime import timedelta
from random import randint

from bsblan import (
    BSBLAN,
    BSBLANAuthError,
    BSBLANConnectionError,
    HotWaterConfig,
    HotWaterSchedule,
    HotWaterState,
    Sensor,
    State,
)

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_HOST
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ConfigEntryAuthFailed
from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed

from .const import DOMAIN, LOGGER, SCAN_INTERVAL_FAST, SCAN_INTERVAL_SLOW


@dataclass
class BSBLanFastData:
    """BSBLan fast-polling data."""

    state: State
    sensor: Sensor
    dhw: HotWaterState


@dataclass
class BSBLanSlowData:
    """BSBLan slow-polling data."""

    dhw_config: HotWaterConfig | None = None
    dhw_schedule: HotWaterSchedule | None = None


class BSBLanCoordinator[T](DataUpdateCoordinator[T]):
    """Base BSB-Lan coordinator."""

    config_entry: ConfigEntry

    def __init__(
        self,
        hass: HomeAssistant,
        config_entry: ConfigEntry,
        client: BSBLAN,
        name: str,
        update_interval: timedelta,
    ) -> None:
        """Initialize the BSB-Lan coordinator."""
        super().__init__(
            hass,
            logger=LOGGER,
            config_entry=config_entry,
            name=name,
            update_interval=update_interval,
        )
        self.client = client


class BSBLanFastCoordinator(BSBLanCoordinator[BSBLanFastData]):
    """The BSB-Lan fast update coordinator for frequently changing data."""

    def __init__(
        self,
        hass: HomeAssistant,
        config_entry: ConfigEntry,
        client: BSBLAN,
    ) -> None:
        """Initialize the BSB-Lan fast coordinator."""
        super().__init__(
            hass,
            config_entry,
            client,
            name=f"{DOMAIN}_fast_{config_entry.data[CONF_HOST]}",
            update_interval=self._get_update_interval(),
        )

    def _get_update_interval(self) -> timedelta:
        """Get the update interval with a random offset.

        Add a random number of seconds to avoid timeouts when
        the BSB-Lan device is already/still busy retrieving data,
        e.g. for MQTT or internal logging.
        """
        return SCAN_INTERVAL_FAST + timedelta(seconds=randint(1, 8))

    async def _async_update_data(self) -> BSBLanFastData:
        """Fetch fast-changing data from the BSB-Lan device."""
        try:
            # Client is already initialized in async_setup_entry
            # Fetch fast-changing data (state, sensor, DHW state)
            state = await self.client.state()
            sensor = await self.client.sensor()
            dhw = await self.client.hot_water_state()

        except BSBLANAuthError as err:
            raise ConfigEntryAuthFailed(
                "Authentication failed for BSB-Lan device"
            ) from err
        except BSBLANConnectionError as err:
            host = self.config_entry.data[CONF_HOST]
            raise UpdateFailed(
                f"Error while establishing connection with BSB-Lan device at {host}"
            ) from err

        # Update the interval with random jitter for next update
        self.update_interval = self._get_update_interval()

        return BSBLanFastData(
            state=state,
            sensor=sensor,
            dhw=dhw,
        )


class BSBLanSlowCoordinator(BSBLanCoordinator[BSBLanSlowData]):
    """The BSB-Lan slow update coordinator for infrequently changing data."""

    def __init__(
        self,
        hass: HomeAssistant,
        config_entry: ConfigEntry,
        client: BSBLAN,
    ) -> None:
        """Initialize the BSB-Lan slow coordinator."""
        super().__init__(
            hass,
            config_entry,
            client,
            name=f"{DOMAIN}_slow_{config_entry.data[CONF_HOST]}",
            update_interval=SCAN_INTERVAL_SLOW,
        )

    async def _async_update_data(self) -> BSBLanSlowData:
        """Fetch slow-changing data from the BSB-Lan device."""
        try:
            # Client is already initialized in async_setup_entry
            # Fetch slow-changing configuration data
            dhw_config = await self.client.hot_water_config()
            dhw_schedule = await self.client.hot_water_schedule()

        except AttributeError:
            # Device does not support DHW functionality
            LOGGER.debug(
                "DHW (Domestic Hot Water) not available on device at %s",
                self.config_entry.data[CONF_HOST],
            )
            return BSBLanSlowData()
        except (BSBLANConnectionError, BSBLANAuthError) as err:
            # If config update fails, keep existing data
            LOGGER.debug(
                "Failed to fetch DHW config from %s: %s",
                self.config_entry.data[CONF_HOST],
                err,
            )
            if self.data:
                return self.data
            # First fetch failed, return empty data
            return BSBLanSlowData()

        return BSBLanSlowData(
            dhw_config=dhw_config,
            dhw_schedule=dhw_schedule,
        )
