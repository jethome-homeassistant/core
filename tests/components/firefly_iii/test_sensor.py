"""Tests for the Firefly III  sensor platform."""

from unittest.mock import AsyncMock, patch

from freezegun.api import FrozenDateTimeFactory
from pyfirefly.exceptions import (
    FireflyAuthenticationError,
    FireflyConnectionError,
    FireflyTimeoutError,
)
import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.firefly_iii.coordinator import DEFAULT_SCAN_INTERVAL
from homeassistant.config_entries import ConfigEntryState
from homeassistant.const import STATE_UNAVAILABLE, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er
from homeassistant.util import dt as dt_util

from . import setup_integration

from tests.common import MockConfigEntry, async_fire_time_changed, snapshot_platform


async def test_all_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_firefly_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    with patch(
        "homeassistant.components.firefly_iii._PLATFORMS",
        [Platform.SENSOR],
    ):
        await setup_integration(hass, mock_config_entry)
        await snapshot_platform(
            hass, entity_registry, snapshot, mock_config_entry.entry_id
        )


@pytest.mark.parametrize(
    ("exception"),
    [
        FireflyAuthenticationError("bad creds"),
        FireflyConnectionError("cannot connect"),
        FireflyTimeoutError("timeout"),
    ],
)
async def test_refresh_exceptions(
    hass: HomeAssistant,
    mock_firefly_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    freezer: FrozenDateTimeFactory,
    exception: Exception,
) -> None:
    """Test entities go unavailable after coordinator refresh failures."""
    await setup_integration(hass, mock_config_entry)
    assert mock_config_entry.state is ConfigEntryState.LOADED

    mock_firefly_client.get_accounts.side_effect = exception

    freezer.tick(DEFAULT_SCAN_INTERVAL)
    async_fire_time_changed(hass, dt_util.utcnow())
    await hass.async_block_till_done()

    state = hass.states.get("sensor.credit_card_account_balance")
    assert state is not None
    assert state.state == STATE_UNAVAILABLE
