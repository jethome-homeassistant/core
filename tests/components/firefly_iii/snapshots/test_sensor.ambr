# serializer version: 1
# name: test_all_entities[sensor.bills_budget-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.bills_budget',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Budget',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'budget',
    'unique_id': 'test_api_key_budget_2_budget',
    'unit_of_measurement': 'AMS',
  })
# ---
# name: test_all_entities[sensor.bills_budget-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'monetary',
      'friendly_name': 'Bills Budget',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': 'AMS',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.bills_budget',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '123.45',
  })
# ---
# name: test_all_entities[sensor.credit_card_account_balance-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.credit_card_account_balance',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Account Balance',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_balance',
    'unique_id': 'test_api_key_account_4_account_balance',
    'unit_of_measurement': 'AMS',
  })
# ---
# name: test_all_entities[sensor.credit_card_account_balance-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'monetary',
      'friendly_name': 'Credit Card Account Balance',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': 'AMS',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.credit_card_account_balance',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '-250.00',
  })
# ---
# name: test_all_entities[sensor.credit_card_account_role-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.credit_card_account_role',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Account Role',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_role',
    'unique_id': 'test_api_key_account_4_account_role',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.credit_card_account_role-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Credit Card Account Role',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.credit_card_account_role',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'creditCard',
  })
# ---
# name: test_all_entities[sensor.credit_card_account_type-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.credit_card_account_type',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:hand-coin',
    'original_name': 'Account Type',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_type',
    'unique_id': 'test_api_key_account_4_account_type',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.credit_card_account_type-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Credit Card Account Type',
      'icon': 'mdi:hand-coin',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.credit_card_account_type',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'liability',
  })
# ---
# name: test_all_entities[sensor.lunch_earned_spent-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.lunch_earned_spent',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Earned/Spent',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'category',
    'unique_id': 'test_api_key_category_2_category',
    'unit_of_measurement': 'AMS',
  })
# ---
# name: test_all_entities[sensor.lunch_earned_spent-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'monetary',
      'friendly_name': 'Lunch Earned/Spent',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': 'AMS',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.lunch_earned_spent',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '-12300.0',
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_balance-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.my_checking_account_account_balance',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Account Balance',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_balance',
    'unique_id': 'test_api_key_account_2_account_balance',
    'unit_of_measurement': 'AMS',
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_balance-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'monetary',
      'friendly_name': 'My checking account Account Balance',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': 'AMS',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_checking_account_account_balance',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '123.45',
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_role-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.my_checking_account_account_role',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Account Role',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_role',
    'unique_id': 'test_api_key_account_2_account_role',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_role-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'My checking account Account Role',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_checking_account_account_role',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'default_asset',
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_type-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.my_checking_account_account_type',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:account-cash',
    'original_name': 'Account Type',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_type',
    'unique_id': 'test_api_key_account_2_account_type',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.my_checking_account_account_type-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'My checking account Account Type',
      'icon': 'mdi:account-cash',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_checking_account_account_type',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'asset',
  })
# ---
# name: test_all_entities[sensor.savings_account_account_balance-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.savings_account_account_balance',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Account Balance',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_balance',
    'unique_id': 'test_api_key_account_3_account_balance',
    'unit_of_measurement': 'AMS',
  })
# ---
# name: test_all_entities[sensor.savings_account_account_balance-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'monetary',
      'friendly_name': 'Savings Account Account Balance',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': 'AMS',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.savings_account_account_balance',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '5000.00',
  })
# ---
# name: test_all_entities[sensor.savings_account_account_role-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.savings_account_account_role',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Account Role',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_role',
    'unique_id': 'test_api_key_account_3_account_role',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.savings_account_account_role-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Savings Account Account Role',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.savings_account_account_role',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'savingsAsset',
  })
# ---
# name: test_all_entities[sensor.savings_account_account_type-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.savings_account_account_type',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:cash-minus',
    'original_name': 'Account Type',
    'platform': 'firefly_iii',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'account_type',
    'unique_id': 'test_api_key_account_3_account_type',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[sensor.savings_account_account_type-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Savings Account Account Type',
      'icon': 'mdi:cash-minus',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.savings_account_account_type',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'expense',
  })
# ---
